controller/volume_attachment_controller_test.go

```diff
@@ -1,400 +1,400 @@
+		"attachment-01": &longhorn.AttachmentTicket{
+			ID:         "attachment-01",
+			Type:       longhorn.AttacherTypeSnapshotController,
+			NodeID:     TestNode1,
+			Parameters: map[string]string{},
+			Generation: 0,
+		},
+		"attachment-02": &longhorn.AttachmentTicket{
+			ID:         "attachment-02",
+			Type:       longhorn.AttacherTypeCSIAttacher,
+			NodeID:     TestNode2,
+			Parameters: map[string]string{},
+			Generation: 0,
+		},
+	}
+	tc.vol.Status.OwnerID = TestNode1
+	tc.vol.Status.State = longhorn.VolumeStateDetached
+	tc.copyCurrentToExpect()
+	tc.expectedVolAttachment.Status.AttachmentTicketStatuses = map[string]*longhorn.AttachmentTicketStatus{
+		"attachment-01": &longhorn.AttachmentTicketStatus{
+			ID:        "attachment-01",
+			Satisfied: false,
+			Conditions: types.SetConditionWithoutTimestamp([]longhorn.Condition{},
+				longhorn.AttachmentStatusConditionTypeSatisfied, longhorn.ConditionStatusFalse, "", ""),
+			Generation: 0,
+		},
+		"attachment-02": &longhorn.AttachmentTicketStatus{
+			ID:        "attachment-02",
+			Satisfied: false,
+			Conditions: types.SetConditionWithoutTimestamp([]longhorn.Condition{},
+				longhorn.AttachmentStatusConditionTypeSatisfied, longhorn.ConditionStatusFalse, "", ""),
+			Generation: 0,
+		},
+	}
+	tc.expectedVol.Spec.NodeID = TestNode2
+	testCases["test case 2: attach: multiple attachments"] = tc
+	///////////////////////////////////////////////////////////////////
+
+	///////////////////////////////////////////////////////////////////
+	tc = generateVolumeAttachmentTestCaseTemplate(TestVolumeName)
+	tc.volAttachment.Spec.AttachmentTickets = map[string]*longhorn.AttachmentTicket{
+		"attachment-01": &longhorn.AttachmentTicket{
+			ID:         "attachment-01",
+			Type:       longhorn.AttacherTypeCSIAttacher,
+			NodeID:     TestNode1,
+			Parameters: map[string]string{},
+			Generation: 0,
+		},
+		"attachment-02": &longhorn.AttachmentTicket{
+			ID:         "attachment-02",
+			Type:       longhorn.AttacherTypeCSIAttacher,
+			NodeID:     TestNode2,
+			Parameters: map[string]string{},
+			Generation: 0,
+		},
+	}
+	tc.vol.Status.OwnerID = TestNode1
+	tc.vol.Status.State = longhorn.VolumeStateDetached
+	tc.copyCurrentToExpect()
+	tc.expectedVolAttachment.Status.AttachmentTicketStatuses = map[string]*longhorn.AttachmentTicketStatus{
+		"attachment-01": &longhorn.AttachmentTicketStatus{
+			ID:        "attachment-01",
+			Satisfied: false,
+			Conditions: types.SetConditionWithoutTimestamp([]longhorn.Condition{},
+				longhorn.AttachmentStatusConditionTypeSatisfied, longhorn.ConditionStatusFalse, "", ""),
+			Generation: 0,
+		},
+		"attachment-02": &longhorn.AttachmentTicketStatus{
+			ID:        "attachment-02",
+			Satisfied: false,
+			Conditions: types.SetConditionWithoutTimestamp([]longhorn.Condition{},
+				longhorn.AttachmentStatusConditionTypeSatisfied, longhorn.ConditionStatusFalse, "", ""),
+			Generation: 0,
+		},
+	}
+	tc.expectedVol.Spec.NodeID = TestNode1
```

Comment:
NBIT: Explain why the expected NodeID is `TestNode1` when the priority is identical. (The candidates are sort by name)

Reply:
Thank you. Fixed